src/api/lookahead.rs

```diff
@@ -145,31 +144,30 @@ pub(crate) fn estimate_inter_costs_histogram_blocks<T: Pixel>(
         height: IMPORTANCE_BLOCK_SIZE,
       });
 
-      let max_value = 1u64 << 12; // At most 12 bit per pixel
-
-      let mut histogram_org =
-        Histogram::configure().max_value(max_value).build().unwrap();
+      let mut count = 0u64;
+      let mut histogram_org_sum = 0u64;
       let iter_org = region_org.rows_iter();
       for row in iter_org {
         for pixel in row {
-          let cur = i16::cast_from(*pixel);
-          histogram_org.increment(cur as u64).unwrap();
+          let cur = u16::cast_from(*pixel);
+          histogram_org_sum += cur as u64;
+          count += 1;
         }
       }
 
-      let mut histogram_ref =
-        Histogram::configure().max_value(max_value).build().unwrap();
+      let mut histogram_ref_sum = 0u64;
       let iter_ref = region_ref.rows_iter();
       for row in iter_ref {
         for pixel in row {
-          let cur = i16::cast_from(*pixel);
-          histogram_ref.increment(cur as u64).unwrap();
+          let cur = u16::cast_from(*pixel);
+          histogram_ref_sum += cur as u64;
         }
       }
 
-      let mean = (histogram_org.mean().unwrap() as i32
-        - histogram_ref.mean().unwrap() as i32)
-        .abs();
+      let mean = ((histogram_org_sum as f64 / count as f64)
```

Comment:
You can do just an integer divide here as well (with adding half of count for rounding).